### PR TITLE
 [TOPIC-BLE-LLCP] bluetooth: controller: llcp: LE PING procedure

### PR DESCRIPTION
**Pull request by erbr-ot**
Implements the le ping procedure and unit tests

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_llcp.h [hunk 1]
```diff
@@ -98,6 +98,11 @@ void ull_cp_run(struct ull_cp_conn *conn);
  */
 void ull_cp_rx(struct ull_cp_conn *conn, struct node_rx_pdu *rx);
 
+/**
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_llcp.h:101]: A thought that occured to me just now: should we re-order these prototypes in the same order as the control procedures are mentioned in the BT spec.?

Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/ull_llcp.h:101]: Fine with me, but it should be done in a separate PR.

File: tests/bluetooth/controller/ctrl_le_ping/src/main.c [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+ * Copyright (c) 2020 Demant
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <ztest.h>
+#include "kconfig.h"
+
+/* Kconfig Cheats */
```
Comment (kruithofa) [inline tests/bluetooth/controller/ctrl_le_ping/src/main.c:11]: We moved the kconfig cheats to the kconfig.h file, please remove comment

File: tests/bluetooth/controller/ctrl_le_ping/src/main.c [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+	/* Rx */
+	lt_tx(LL_LE_PING_RSP, &conn, &remote_ping_rsp);
+
+	/* Done */
+	event_done(&conn);
+
+	/* There should not be a host notifications */
+	ut_rx_q_is_empty();
+}
```
Comment (kruithofa) [inline tests/bluetooth/controller/ctrl_le_ping/src/main.c:98]: Strictly spoken we should release the tx buffer here by a call to ull_cp_release_tx(tx)

File: tests/bluetooth/controller/ctrl_le_ping/src/main.c [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+	/* There should not be a host notifications */
+	ut_rx_q_is_empty();
+
+}
+
+void test_main(void)
+{
+	ztest_test_suite(ping,
+             ztest_unit_test_setup_teardown(test_ping_mas_loc, setup, unit_test_noop),
+			 ztest_unit_test_setup_teardown(test_ping_sla_loc, setup, unit_test_noop),
```
Comment (kruithofa) [inline tests/bluetooth/controller/ctrl_le_ping/src/main.c:260]: indentation is wrong

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_llcp.h (hunk 1 above)
Comment (kruithofa) [reply subsys/bluetooth/controller/ll_sw/ull_llcp.h:101]: Agreed

**Comment by erbr-ot** [comment]
@kruithofa Thanks Andries - all comments addressed. I decided to remove the decode_ping... functions all together and moved the comment /* ... has not data */ to the switch case instead. I hope you agree.

**Review by kruithofa — APPROVED** [review]
LGTM

**Pull request merged**